src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -128,10 +128,29 @@ private CSharpSyntaxNode ParseTypeOrPatternForIsOperatorCore()
         /// </summary>
         private bool LooksLikeTypeOfPattern(SyntaxKind tk)
```

Comment:
>LooksLikeTypeOfPattern [](start = 21, length = 22)

It feels strange that part of this function uses parameter ```tk``` to figure out the answer and the other part uses current position in the token stream. This can confuse consumers thinking that it is enough to pass the ```SyntaxKind``` and the current position doesn't matter. Consider removing the parameter.